### ISSUE TITLE
fix(blink): optimize calling

### ISSUE THREAD

**Issue by sdwoodbury** [Satellite-im/Warp#366]
# todo
- use a fixed buffer size for exchanging data with CPAL. Blink currently uses whatever the hardware defaults to, and if the buffer is very large this could cause latency issues. 
- minimize the number of calls to `recv()` inside of the CPAL callbacks. Currently Blink is using a queue of samples but instead it should use a queue of Vec<u8> to send an entire opus packet to CPAL all at once. 
- perform opus encoding and decoding on dedicated threads. Currently Blink spawns a task for each peer connection. I suspect the audio encoding/decoding would be better if done in their own `std::thread` instead of a tokio task. Decoding of peer's audio could be done in parallel using `rayon`. 
- already done but not merged in yet: change the signaling to be more robust. Peers can "announce" themselves on a gossipsub channel and then connect to each other based off of those signals. The "offer" signal should also be periodically retried upon failure. This will fix an issue with group calls where peers don't always connect to each other because they hadn't discovered each other at the time the call was offered. 

# additional areas of improvement
- the abstraction to allow different audio codecs seems unnecessary at this time. 
- the abstraction to allow different audio sample rates also seems unnecessary. The opus codec converts audio to 48khz internally anyway and I suspect all modern hardware can support this.

**Comment by sdwoodbury** [comment]
done
